docs/src/tutorials/discrete_stochastic_example.md

```diff
@@ -371,60 +472,69 @@ both must be constant other than changes due to some other `ConstantRateJump` or
+\beta S(t) I(t) + \alpha S(t) \sum_{t_i \in H(t)} \exp(-\gamma (t - t_i))
 ```
-Where ``\beta`` is the basal rate of infection, ``\alpha`` is the spike in the
-rate of infection and ``\gamma`` is the rate which the spike decreases. Here we
-choose parameters such that infectivity reaches a basal rate close to ``0``
-after spiking. The spike is equal to the rate ``\beta`` chosen in the previous
-section and ``gamma`` is the same as the recovery rate. In other words, we are
-modelling a situation in which individuals gradually recover.
+where ``\beta`` is the basal rate of infection, ``\alpha`` is the spike in the
+rate of infection, and ``\gamma`` is the rate at which the spike decreases. Here
+we choose parameters such that infectivity rate due to a single infected
+individual returns to the basal rate after spiking to ``\beta + \alpha``. In
+other words, we are modelling a situation in infected individuals gradually
+become less infectious prior to recovering. Our parameters are then
 
 ```@example tut2
 β1 = 0.001 / 1000.0
 α = 0.1 / 1000.0
-γ = 0.001
+γ = 0.05
 p1 = (β1, ν, α, γ)
 ```
 
-We define a vector `H` to hold the timestamp of active infections. Then, we 
-define infection as a `VariableRateJump`. To use the `Coevolve` aggregator, 
-we need to specify the lower- and upper-bounds of the rate which should be valid 
-from the time they are computed `t` until `t + L(u, p, t)`.
+We define a vector `H` to hold the timestamp of active infections. Then, we
+define an infection reaction as a bounded `VariableRateJump`, requiring us to
+again provide `rate` and `affect` functions, but also give functions that
+calculate an upper-bound on the rate (`urate(u,p,t)`), an optional lower-bound
+on the rate (`lrate(u,p,t)`), and a time window over which the bounds are valid
+as long as any states these three rates depend on are unchanged
+(`rateinterval(u,p,t)`).
+The lower- and upper-bounds of the rate should be valid from the time they are
+computed `t` until `t + rateinterval(u, p, t)`:
 
 ```@example tut2
 H = zeros(Float64, 10)
-rate3(u, p, t) = p[1]*u[1]*u[2] + p[3]*u[1]*sum([exp(-p[4]*(t - _t)) for _t in H])
+rate3(u, p, t) = p[1]*u[1]*u[2] + p[3]*u[1]*sum(exp(-p[4]*(t - _t)) for _t in H)
 lrate = rate1              # β*S*I
 urate = rate3
-L(u, p, t) = 1 / (2*urate(u, p, t))
+rateinterval(u, p, t) = 1 / (2*urate(u, p, t))
 function affect3!(integrator)
   integrator.u[1] -= 1     # S -> S - 1
   integrator.u[2] += 1     # I -> I + 1
   push!(H, integrator.t)
   nothing
 end
-jump3 = VariableRateJump(rate3, affect3!; lrate=lrate, urate=urate, L=L)
+jump3 = VariableRateJump(rate3, affect3!; lrate, urate, rateinterval)
 ```
+Note that here we set the lower bound rate to be the normal SIR infection rate,
+and set the upper bound rate equal to the new rate of infection (`rate3`). As
+long as `u[1]` and `u[2]` are unchanged by another jump, for any `s` in `[t,t +
+rateinterval(u,p,t)]` we have that `lrate(u,p,t) <= rate3(u,p,s) <= urate(u,p,t)`.
```

Comment:
I would add an observation that the rates will be re-computed when `u[1]` and `u[2]` change to avoid the reader assuming that `u[1]` or `u[2]` cannot change.
```suggestion
Given `u[1]` and `u[2]`, for any `s` in `[t, t + rateinterval(u,p,t)]` we have 
that `lrate(u,p,t) <= rate3(u,p,s) <= urate(u,p,t)`. Note that the bounds are
automatically recomputed when `u[1]` or `u[2]` changes using the dependency 
graph supplied to `Coevolve`.
```